src/main/java/com/google/opensesame/user/UserData.java

```diff
@@ -0,0 +1,138 @@
+package com.google.opensesame.user;
+
+import com.google.opensesame.auth.AuthServlet;
+import com.google.opensesame.github.GitHubGetter;
+import java.io.IOException;
+import java.util.ArrayList;
+import org.kohsuke.github.GHUser;
+import org.kohsuke.github.GitHub;
+
+/**
+ * A fully fleshed out User of OpenSesame containing information sourced from GitHub. To conserve
+ * API calls, It would be best to only build UserData where neccesary. The main use of this class is
+ * to send data to the frontend as a JSON with Gson.
+ */
+public class UserData {
+  private Boolean isMentor;
+  private final ArrayList<String> interestTags;
+  private final ArrayList<String> menteeIDs;
+  private final ArrayList<String> projectIDs;
+  private final String bio;
+  private final String gitHubID;
+  private final String gitHubURL;
+  private final String image;
+  private final String location;
+  private final String name;
+  private final String userID;
+  private String email;
+
+  /**
+   * Create a UserData from a given UserEntity
+   *
+   * <p>Note: Emails will only ever be stored in a UserObject if the current user is authorized with
+   * the AuthServlet.
+   *
+   * @param userEntity
+   * @throws IOException
+   */
+  public UserData(UserEntity userEntity) throws IOException {
+    this.gitHubID = userEntity.gitHubId;
+    this.interestTags = userEntity.interestTags;
+    this.userID = userEntity.userId;
+    this.email = userEntity.email;
+    // Only ever send email stuff when a user is authorized
+    if (AuthServlet.getAuthorizedUser() == null) {
+      this.email = null;
+    }
+    this.projectIDs = userEntity.projectIDs;
+    this.menteeIDs = userEntity.menteeIDs;
+    this.isMentor = userEntity.isMentor;
+
+    // GitHub query TODO: alternate ways of getting this information.
```

Comment:
Was this what we discussed earlier today? Should we consider this resolved?

Reply:
No, this is for something different. I'll make the TODO more descriptive.